src/internal/analytics-metadata/attributes.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { GeneratedAnalyticsMetadataFragment } from './interfaces';
+
+export const METADATA_DATA_ATTRIBUTE = 'awsuiAnalytics';
+export const METADATA_ATTRIBUTE = `data-awsui-analytics`;
+export const LABEL_DATA_ATTRIBUTE = 'awsuiAnalyticsLabel';
+const LABEL_ATTRIBUTE = 'data-awsui-analytics-label';
+
+export const getAnalyticsMetadataAttribute = (metadata: GeneratedAnalyticsMetadataFragment) => ({
+  [METADATA_ATTRIBUTE]: JSON.stringify(metadata),
+});
+
+export const copyAnalyticsMetadataAttribute = (props: any) => ({
+  [METADATA_ATTRIBUTE]: props[METADATA_ATTRIBUTE],
```

Comment:
I can't see where this is being used. Is it copying string props or could it be object props? If objects then this might be a problem as it returns a shallow copy.

Reply:
It is useful when dealing with rest properties, to avoid injecting other properties that would be passed, and just picking the data attribute for analytics. See, for example, https://github.com/cloudscape-design/components/commit/35ea770e181a9187c2b6226e8467a82bc9ddc944#diff-a52cdfeda9c54f63d9558a83e9c81a720fd1cbc32fc2ec1136e6e825816e8be0R221